include/json.h

```diff
@@ -110,7 +116,7 @@ typedef int (*json_append_bytes_t)(const char *bytes, size_t len,
 		.field_name = (#field_name_), \
 		.field_name_len = sizeof(#field_name_) - 1, \
 		.offset = offsetof(struct_, field_name_), \
-		.alignment = __alignof__(struct_), \
+		.alignment = __alignof__(struct_) - 1, \
```

Comment:
Why do we need to have -1 here?

Reply:
Added a comment in struct json_obj_descr explaining the reasoning.  Went ahead and added some BUILD_ASSERT macros as well to ensure that these values are sane.